include/mbgl/style/style.hpp

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include <mbgl/style/transition_options.hpp>
+#include <mbgl/util/geo.hpp>
+
+#include <string>
+#include <vector>
+#include <memory>
+
+namespace mbgl {
+
+class FileSource;
+class Scheduler;
+
+namespace style {
+
+class Light;
+class Image;
+class Source;
+class Layer;
+
+class Style {
+public:
+    Style(Scheduler&, FileSource&, float pixelRatio);
+    ~Style();
+
+    void loadJSON(const std::string&);
+    void loadURL(const std::string&);
+
+    std::string getJSON() const;
+    std::string getURL() const;
+
+    // Defaults
+    std::string getName() const;
+    LatLng getDefaultLatLng() const;
+    double getDefaultZoom() const;
+    double getDefaultBearing() const;
+    double getDefaultPitch() const;
+
+    // TransitionOptions
+    TransitionOptions getTransitionOptions() const;
+    void setTransitionOptions(const TransitionOptions&);
+
+    // Light
+          Light* getLight();
```

Comment:
Should we use indentation here that is more common? Not sure if this helps with readability, but it prevents using standard formatting tools.